tools/tdbg/commands.go

```diff
@@ -371,21 +371,51 @@ func AdminGetShardID(c *cli.Context) error {
 	return nil
 }
 
+// getCategory first searches the registry for the category by the [tasks.Category.Name]. If not found, it checks
+// against a hard-coded mapping of name to category which was defined back when the TaskCategory was an enum. It used to
+// be an enum because we didn't support dynamically-registered task categories, but now we do, so it needs to be an int
+// instead of an enum. It used to look like this:
+//
+//	var TaskCategory_value = map[string]int32{
+//		 "Unspecified": 0,
+//		 "Transfer":    1,
+//		 "Timer":       2,
+//		 "Replication": 3,
+//		 "Visibility":  4,
+//		 "Archival":    5,
+//		 "MemoryTimer": 6,
+//	}
+func getCategory(registry tasks.TaskCategoryRegistry, key string) (tasks.Category, error) {
+	for _, category := range registry.GetCategories() {
+		if category.Name() == key {
+			return category, nil
+		}
+	}
+	switch key {
+	case "Transfer":
+		return tasks.CategoryTransfer, nil
+	case "Timer":
+		return tasks.CategoryTimer, nil
+	case "Replication":
+		return tasks.CategoryReplication, nil
+	case "Visibility":
+		return tasks.CategoryVisibility, nil
+	case "Archival":
+		return tasks.CategoryArchival, nil
+	case "MemoryTimer":
+		return tasks.CategoryTimer, nil
+	default:
+		return tasks.Category{}, fmt.Errorf("unknown task category %q", key)
+	}
```

Comment:
is this just for backward compatibility? Feels unnecessary to me since only admin/tdbg commands are using it.

Reply:
Yeah, that's true. Removed